raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java

```diff
@@ -199,6 +206,34 @@ private static Map<Integer, InetSocketAddress> parseVoterConnections(
         return voterMap;
     }
 
+    public static InetSocketAddress parseBootstrapServer(String bootstrapServer) {
+        String host = Utils.getHost(bootstrapServer);
+        if (host == null) {
+            throw new ConfigException(
+                String.format(
+                    "Failed to parse host name from {} for the configuration {}. Each " +
+                    "entry should be in the form \"{host}:{port}\"",
+                    bootstrapServer,
+                    QUORUM_BOOTSTRAP_SERVERS_CONFIG
+                )
+            );
+        }
+
+        Integer port = Utils.getPort(bootstrapServer);
+        if (port == null) {
+            throw new ConfigException(
+                String.format(
+                    "Failed to parse host port from {} for the configuration {}. Each " +
+                    "entry should be in the form \"{host}:{port}\"",
+                    bootstrapServer,
+                    QUORUM_BOOTSTRAP_SERVERS_CONFIG
+                )
+            );
+        }
+
+        return InetSocketAddress.createUnresolved(host, port);
```

Comment:
I am not sure. I know that it throws an `IllegalArgumentException` if host is `null` or port is outside the valid range. I think it is better for `Utils.getHost` and `Utils.getPort` to better catch invalid host and port values so we can throw the corresponding `ConfigException`. I filed https://issues.apache.org/jira/browse/KAFKA-16824 to improve the implementation of those methods.

Reply:
I think it's a mistake to resolve hostnames as part of configuration validation. Just treat them as strings.

Otherwise you could end up not being able to start Kafka because 1 out of 3 hostnames in your bootstrap list could not be resolved (because of a DNS problem or whatever)

Resolve a hostname when you're actually using it, not before.

Reply:
I agree. I have been fixing any code that resolves host names outside of the `NetworkClient`. This PR should not be resolving host names outside of the `NetworkClient`. Let me know if you find a case that violates this.